### PR TITLE
only apply timeouts if running in Dynamic SKU

### PR DESCRIPTION
**Pull request by brettsam**

### CODE REVIEW

File: src/WebJobs.Script/Host/ScriptHost.cs [hunk 1]
```diff
@@ -841,7 +846,21 @@ internal static void ApplyConfiguration(JObject config, ScriptHostConfiguration
...
+                    string message = $"{nameof(scriptConfig.FunctionTimeout)} must be between {MinTimeout} and {MaxTimeout}.";
+                    throw new ArgumentException(message);
+                }
+
+                scriptConfig.FunctionTimeout = requestedTimeout;
+            }
+            else if (IsDynamicSku())
+            {
+                // Apply a default if this is running on Dynamic.
+                scriptConfig.FunctionTimeout = TimeSpan.FromMinutes(5);
```
Comment (mathewc) [inline src/WebJobs.Script/Host/ScriptHost.cs]: Use MaxTimeout here?

File: src/WebJobs.Script/Host/ScriptHost.cs [hunk 2]
```diff
@@ -841,7 +846,21 @@ internal static void ApplyConfiguration(JObject config, ScriptHostConfiguration
...
             if (config.TryGetValue("timeout", out value))
             {
-                scriptConfig.FunctionTimeout = TimeSpan.Parse((string)value, CultureInfo.InvariantCulture);
+                TimeSpan requestedTimeout = TimeSpan.Parse((string)value, CultureInfo.InvariantCulture);
+
+                // Only apply limits if this is Dynamic.
+                if (IsDynamicSku() && (requestedTimeout < MinTimeout || requestedTimeout > MaxTimeout))
+                {
+                    string message = $"{nameof(scriptConfig.FunctionTimeout)} must be between {MinTimeout} and {MaxTimeout}.";
+                    throw new ArgumentException(message);
```
Comment (mathewc) [inline src/WebJobs.Script/Host/ScriptHost.cs]: Should this be ArgumentOutOfRangeException?

**Comment by mathewc** [comment]
:ship:

File: src/WebJobs.Script/Host/ScriptHost.cs (hunk 2 above)
Comment (brettsam) [inline src/WebJobs.Script/Host/ScriptHost.cs]: That's what I had in my original commit and @fabiocav suggested an ArgumentException. I could go either way.

---
In reply to: [78202291](https://github.com/Azure/azure-webjobs-sdk-script/pull/661#discussion_r78202291) [](ancestors = 78202291)

**Pull request merged**